### PR TITLE
Change farmland mixins

### PR DESCRIPTION
**Pull request by Jack-Papel**
These now will not conflict with other mods' mixins, and also will not silently fail.
These mixins will even still work if other mod makers are incompetent, because these mixins will be applied last.

### CODE REVIEW

File: src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/CropBlockMixin.java [hunk 1]
```diff
@@ -46,10 +46,10 @@ private static boolean isBlockFarmland(BlockState blockState, Block block) {
     
     @Inject(
         method = "canPlantOnTop",
-        at = @At("HEAD"),
+        at = @At("RETURN"),
         cancellable = true
     )
     private void canPlantOnTop(BlockState floor, BlockView world, BlockPos pos, CallbackInfoReturnable<Boolean> cir) {
-        cir.setReturnValue(floor.isIn(MoreBlockTags.FARMLAND));
+        cir.setReturnValue(cir.getReturnValue() || floor.isIn(MoreBlockTags.FARMLAND));
```
Comment (Pyrofab) [inline src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/CropBlockMixin.java:53]:
```suggestion
if (!cir.getReturnValue() && floor.isIn(MoreBlockTags.FARMLAND)) {
        cir.setReturnValue(true);
}
```

File: src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/PlantBlockMixin.java [hunk 2]
```diff
@@ -1,29 +1,23 @@
...
-        at = @At(
-            value = "INVOKE",
-            target = "Lnet/minecraft/block/BlockState;isOf(Lnet/minecraft/block/Block;)Z"
-        ),
+        at = @At("RETURN"),
         cancellable = true
     )
     private void canPlantOnTop(BlockState floor, BlockView world, BlockPos pos, CallbackInfoReturnable<Boolean> cir) {
-        cir.setReturnValue(floor.isIn(MoreBlockTags.PLANTABLE_BLOCKS));
+        cir.setReturnValue(cir.getReturnValue() || floor.isIn(MoreBlockTags.FARMLAND));
```
Comment (Pyrofab) [inline src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/PlantBlockMixin.java:21]: Is the change from plantable_blocks to farmland intended here ?

File: src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/StemBlockMixin.java [hunk 3]
```diff
@@ -13,15 +13,15 @@
...
 public abstract class StemBlockMixin {
     @Inject(
         method = "canPlantOnTop",
-        at = @At("HEAD"),
+        at = @At("RETURN"),
         cancellable = true
     )
     private void canPlantOnTop(BlockState floor, BlockView world, BlockPos pos, CallbackInfoReturnable<Boolean> cir) {
-        cir.setReturnValue(floor.isIn(MoreBlockTags.FARMLAND));
+        cir.setReturnValue(cir.getReturnValue() || floor.isIn(MoreBlockTags.FARMLAND));
```
Comment (Pyrofab) [inline src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/StemBlockMixin.java:24]:
```suggestion
if (!cir.getReturnValue() && floor.isIn(MoreBlockTags.FARMLAND)) {
        cir.setReturnValue(true);
}
```

**Review by Pyrofab — COMMENTED** [review]
Looks like an important change, but I believe we can make it a bit more compatible still

File: src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/PlantBlockMixin.java (hunk 2 above)
Comment (Jack-Papel) [reply src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/PlantBlockMixin.java:21]: Unintended. Good catch

File: src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/CropBlockMixin.java (hunk 1 above)
Comment (Jack-Papel) [reply src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/CropBlockMixin.java:53]: Is this to let other mods deny certain blocks?

Comment (gudenau) [reply src/main/java/net/gudenau/minecraft/moretags/mixins/block/farmland/CropBlockMixin.java:53]: That would prevent you from removing farmland blocks.

**Comment by gudenau** [comment]
These changes change the behavior of the mod.

**Comment by Jack-Papel** [comment]
Indeed. They make the mod more compatible with other mods. That is their purpose

**Comment by Jack-Papel** [comment]
Oh, wait. I see your concerns. These changes make it so that you can't remove farmland blocks from the tag

**Comment by gudenau** [comment]
Exactly the problem, the reason I use tags for this is to allow data pack authors to remove functionality from blocks.

**Comment by Jack-Papel** [comment]
Understood.

**Comment by Zifiv** [comment]
So you prefere break other mods than being compatible... nice choice...

**Pull request closed without merging**